Removed unnecessary dependencies

juanjalvarez opened the pull request:
The is-svg package is not consistent at svg detection (mentioned on their readme). We do not really need an external library for this since the file object is already sent to the contentType callback along with it's mimetype. With this PR the npm package goes down from 12.6kb to 1.2kb.

LinusU commented:
There is a security risk involved with trusting the mime type that the browser sends, the client can be manipulated to send whatever mime type they want...

juanjalvarez commented:
Would you be open to considering adding a second autoContentType, maybe named contentTypePassthrough that allows passing the given mimetype?

Also, we could just get rid of specifically is-svg and replace it with libxmljs (is-svg recommends this for accurate svg detection) and only use it if the path ends in .svg since libxmljs requires you to read the entire file.

For context the underlying issue I am trying to address is that I ran into an svg file that this library is failing to identify as valid svg and the failure comes from is-svg. In my case I just passed the contentType callback a function that does exactly as what is in the PR, but if I had this issue it is possible that other people will too.

LinusU commented:
I think that the problem comes from only passing the first chunk to `is-svg`. In my opinion we could use a simple regex like this:

```js
/^\s*(?:<\?xml[^>]*>\s*)?(?:<!doctype svg[^>]*\s*(?:\[?(?:\s*<![^>]*>\s*)*\]?)*[^>]*>\s*)?<svg[^>]*>/i
```

If something starts out as a valid svg document, but doesn't end with `</svg>` it was probably still meant to be an svg file. The other file types are being detected based on some magic bytes in the beginning of the file, so there has never been any indication that it should be a valid file from any of the detected types...

juanjalvarez commented:
That was a pretty good idea! I have updated the PR to use that instead, also removed is-svg. I made sure that if there is at least one match with that regex then it should be valid svg. Tests pass with the current svg and my previously failing svg.

LinusU commented on index.js:
```diff
@@ -30,10 +31,11 @@ function autoContentType (req, file, cb) {
   file.stream.once('data', function (firstChunk) {
     var type = fileType(firstChunk)
     var mime
+    var svgMatches = firstChunk.toString().match(svgRegex) || []
```
Use `.test` to get a boolean instead of an array ☺️ 

```suggestion
    var svgMatches = svgRegex.test(firstChunk.toString())
```

LinusU replied:
does `.toString()` always work on a `Buffer`, even if it contains invalid unicode codes? 🤔

juanjalvarez replied:
It passes on the PNG test, when it converts the PNG buffer to string it just contains invalid characters.

juanjalvarez replied:
![image](https://user-images.githubusercontent.com/8794965/54306659-1b27b780-4587-11e9-9a0a-471df948f855.png)

LinusU commented:
@badunk would you like to review this as well? ☺️
